### PR TITLE
Bug fixes for array subtyping at returns / parameter passing

### PR DESCRIPTION
**Pull request by msridhar**
In certain places we were bailing out for types without type arguments, but those bailouts are wrong now that we check array subtyping.  Also, for identifiers, we now get the type from the `Symbol` which is more reliable.

### CODE REVIEW

File: nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java [hunk 1]
```diff
@@ -278,6 +278,51 @@ public void arraysAndGenerics() {
...
+            "    }",
+            "    void takeFoos(Foo<T>[] foos) {}",
+            "    void callTakeFoosPositive(@Nullable Foo<T>[] p) {",
+            "      // BUG: Diagnostic contains: Cannot pass parameter of type @Nullable Foo<T>[]",
+            "      takeFoos(p);",
+            "    }",
+            "    void callTakeFoosNegative(Foo<T>[] p) {",
+            "      takeFoos(p);",
+            "    }",
+            "    void takeFoosVarargs(Foo<T>[]... foos) {}",
```
Comment (lazaroclapp) [inline nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java:308]: To be clear, the type of `foos` here ends up being `Foo<T>[][]` right? With a single element array passed in 311 that gets converted to the type `@Nullable Foo<T>[][]` (meaning a two dimensional array of nullable `Foo`s), correct?

File: nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java [hunk 2]
```diff
@@ -278,6 +278,51 @@ public void arraysAndGenerics() {
...
+            "      // BUG: Diagnostic contains: Cannot pass parameter of type @Nullable Foo<T>[]",
+            "      takeFoosVarargs(p);",
+            "    }",
+            "    void callTakeFoosVarargsNegative(Foo<T>[] p) {",
+            "      takeFoosVarargs(p);",
+            "    }",
+            "    void takeNullableFoosVarargs(@Nullable Foo<T>[]... foos) {}",
+            "    void callTakeNullableFoosVarargsNegative(@Nullable Foo<T>[] p1, Foo<T>[] p2) {",
+            "      takeNullableFoosVarargs(p1);",
+            "      takeNullableFoosVarargs(p2);",
```
Comment (lazaroclapp) [inline nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java:319]: What about: `takeNullableFoosVarargs(p1, p2)`? (also worth having the case for  `takeFoosVarargs(p2, p1)`, I guess)

**Review by lazaroclapp — APPROVED** [review]
A couple test case suggestions, but overall LGTM

File: nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java (hunk 1 above)
Comment (msridhar) [reply nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java:308]: Yes, that's right; added a comment in 4aa9da7435062a0e308009ba76a8fa61b47a2a01.  Note that we don't have support for multidim arrays yet in JSpecify mode, so we can't yet test passing in a `@Nullable Foo<T>[][]` array directly.

File: nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java (hunk 2 above)
Comment (msridhar) [reply nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java:323]: Good idea, added in 26b1eaf8c93a3b6f3a80d7a5f128f2806a4bad35

**Pull request merged**